### PR TITLE
[WIP] Enable  rovers & boats to slow for nav turns, multicopter-style

### PR DESCRIPTION
**Pull request by packetpilot**
The only substantive piece of this PR is lifted directly from the multicopter flavor of this navigation source, in order to let lowly and often dirty surface vehicles run WP missions without either doing so at a snail's pace (boring!) or roll over at the first turn.

Big thanks to @sensei-hacker for pointing me in the right direction in the discord "server" (a word discord misuses egregiously -- sorry, back to the point).

Other minutiae relates to branding/spelling/ordering/testing -- speaking of testing, I did run the tests, and did not know whether this change warranted any new test creation since no code is "new" -- that said, I've literally never written, let alone manipulated/adapted, a single line of C before  -- only build pipelines and infra around the C itself -- so _please_ look carefully to see if I've done something very dumb, because I'm just an sysadmin turned ops plumber w/ skills mainly in "boring" stuff like CI/containers/infra.

In terms of _meatspace_, testing has been carried out on two Matek FCs, an F722-WPX and an old F411WING.

Hopes are high that I'll make more rover-oriented contributions in the future so course-correction is very welcome and so is direct criticism, so long as it's constructive.

### CODE REVIEW

**Comment by packetpilot** [comment]
Just got done w/ some sanity checks (byproduct of steeping in self-doubt) and realized "something must've changed" between the 7.0.0 tag and the current state of 7.1 release branch that made this (or, the original) pull request moot. About to see if I've figured it out. Apologies for the premature PR; [WIP] prefix on title will be removed once I commit a change that's verified working against the target branch.

**Comment by packetpilot** [comment]
TL;DR probably the new `nav_min_ground_speed` in my way, will try to address that without doing anything "dumb" (i.e. making changes that'd be able to impact actual aircraft)
---
Connector woes on my three FCs (-10..+10°C temps don't help!) slowed my path to enlightenment via `git diff 7.0.0..release_7.1.0 docs/Settings.md` ... my gut says the new `nav_min_ground_speed` parameter is doing its job & preventing me from observing the expected behavior. Guess it's time to lean on logs and not just LoS observation.

With zero C chops, I'll see if I can sort out how to remove that obstacle, reasonably (and for rovers/boats _only_), since I don't live on a skidpad, and have no friends who live at the "hangar houses" nearby -- two "driveways" per house and the one out back is a small airstrip's taxiway! The stuff of dreams!!

Should I resort to driving a "rover that looks like a flying wing" around on the ground in xplane, please slam the door on this pull & issue permaban. Joking aside, I do wonder if getting a HIL/SItL setup with VRC Pro (or even AccuRC -- which has a few cars) would be -- scratch that, if I want to rely on sim I can just use RF and remove control surfaces and airfoils on something with a tricycle gear.

If anyone cares to nudge me in one direction or another, here or discord works -- over there you'll spot me among the spades of lurkbots only there to pipe what's said into NLP datasets...

**Comment by MrD-RC** [comment]
Just a note on `nav_min_ground_speed`. The mechanism to maintain a minimum ground speed has been in INAV for a long time. It used to be hard coded to 7m/s. The new `nav_min_ground_speed` parameter just allows the speed to be changed. Rather than it being hard coded.

Also, the parameter is only used in `navigation_fixedwing.c`. So unless boats and rovers are using this file as their navigation controller. It should not be affecting them.  You can see the code [here](https://github.com/iNavFlight/inav/pull/9428/files). I can see a `navigation _rover_boat.c` file. Which has no mention of minimum ground speeds. So the problem must be something else.

**Comment by sensei-hacker** [comment]
> Just got done w/ some sanity checks (byproduct of steeping in self-doubt) and realized "something must've changed" between the 7.0.0 tag and the current state of 7.1 release branch that made this (or, the original) pull request moot.

If I were in your shoes, I would double check that it works in 7.0.0, then try a commit that's roughly halfway in between.  Testing just two points in between cuts the number of potentially conflicting changes to look at by 75%.

**Comment by packetpilot** [comment]
Good suggestion. I'm going to close this PR and get my house in order, and
hopefully before too long (i.e. before 7.1 freeze) have a new PR. I
admittedly wasn't very methodical and the ephemeral envs I'd been working
in were hastily nuked.

I'll be more careful before opening a pull next time. Thanks folks for the
pointers.
